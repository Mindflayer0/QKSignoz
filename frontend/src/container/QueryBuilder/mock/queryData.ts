import { initialQueryBuilderFormValuesMap } from 'constants/queryBuilder';
import { IBuilderQuery } from 'types/api/queryBuilder/queryBuilderData';
import { DataSource } from 'types/common/queryBuilder';

export const queryMockData: IBuilderQuery = {
	dataSource: DataSource.METRICS,
	queryName: 'A',
	aggregateOperator: 'noop',
	aggregateAttribute: {
		key: 'signoz_latency_count',
		dataType: 'float64',
		type: 'tag',
		isColumn: true,
	},
	filters: {
		items: [
			{
				id: '8fac746b',
				key: {
					key: 'resource_signoz_collector_id',
					dataType: 'int64',
					isColumn: false,
					type: 'tag',
				},
				op: '=',
				value: ['1a5d3cc2-4b3e-4c7c-ad07-c4cdd739d1b9'],
			},
			{
				id: '8fadtr46b',
				key: {
					key: 'service_name',
					dataType: 'string',
					isColumn: false,
					type: 'tag',
				},
				op: 'IN',
				value: ['frontend', 'operation', 'ser'],
			},
		],
		op: 'AND',
	},
	expression: '',
	disabled: false,
	having: [],
	stepInterval: 30,
	limit: 10,
	orderBy: [],
	groupBy: [],
	legend: '',
	reduceTo: 'sum',
};

export const formulaMockData = {
	queryName: 'C',
	label: 'Formula',
	expression: 'sum(A, B)',
	legend: 'Total',
	disabled: false,
};

export const valueWithAttributeAndOperator: IBuilderQuery = {
	...initialQueryBuilderFormValuesMap.logs,
	dataSource: DataSource.LOGS,
	aggregateOperator: 'SUM',
	aggregateAttribute: {
		isColumn: false,
		key: 'bytes',
		type: 'tag',
		dataType: 'float64',
	},
};

export const querySearchResult = ['frontend', 'operation', 'ser'];
