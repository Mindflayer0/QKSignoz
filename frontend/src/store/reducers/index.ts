import { combineReducers } from 'redux';

import appReducer from './app';
import dashboardReducer from './dashboard';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { PipelineReducer } from './pipeline';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	dashboards: dashboardReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
	pipeline: PipelineReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
