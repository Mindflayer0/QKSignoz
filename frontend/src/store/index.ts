import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
