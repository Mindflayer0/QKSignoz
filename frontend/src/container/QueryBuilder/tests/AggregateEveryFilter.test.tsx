import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import AggregateEveryFilter from '../filters/AggregateEveryFilter';
import { queryMockData } from '../mock/queryData';

describe('QueryBuilder: Aggregate Every Filter', () => {
	it('it should render AggregateEveryFilter', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<AggregateEveryFilter onChange={jest.fn()} query={queryMockData} />
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
