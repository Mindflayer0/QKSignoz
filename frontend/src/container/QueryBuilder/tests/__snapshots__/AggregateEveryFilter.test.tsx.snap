// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: Aggregate Every Filter it should render AggregateEveryFilter 1`] = `
<DocumentFragment>
  <input
    class="ant-input css-dev-only-do-not-override-1i536d8"
    placeholder="Enter in seconds"
    style="width: 100%; min-width: 7.7rem;"
    type="text"
    value="30"
  />
</DocumentFragment>
`;
